### ISSUE TITLE
Exercise 4 Level 6

### ISSUE THREAD

**Issue by shuyaoao** [nus-cs2030/2223-s2#475]
Hiii anyone has done exercise 4 level 6? What approach did yall take? Do we have to create a interface for Maybe?

**Comment by 0liverkong** [comment]
yup. follow the lecture on the maybe context. you should only have one file, Maybe.java

**Comment by joshuagohez** [comment]
> yup. follow the lecture on the maybe context. you should only have one file, Maybe.java

did you have to redefine the private method signatures of get(), isEmpty() etc to cater for Maybe<?>  and Maybe<? extends T>?

**Comment by joshuagohez** [comment]
> > yup. follow the lecture on the maybe context. you should only have one file, Maybe.java
> 
> did you have to redefine the private method signatures of get(), isEmpty() etc to cater for Maybe<?> and Maybe<? extends T>?

Solved. Alls good.

**Comment by lsy8117** [comment]
> > > yup. follow the lecture on the maybe context. you should only have one file, Maybe.java
> > 
> > 
> > did you have to redefine the private method signatures of get(), isEmpty() etc to cater for Maybe<?> and Maybe<? extends T>?
> 
> Solved. Alls good.

How did you managed to solve this? i'm facing this issue right now

**Comment by 0liverkong** [comment]
> > yup. follow the lecture on the maybe context. you should only have one file, Maybe.java
> 
> did you have to redefine the private method signatures of get(), isEmpty() etc to cater for Maybe<?> and Maybe<? extends T>?

for others who are curious, you shouldnt need to define multiple types of get() or isEmpty(). everything should be within the anonymous inner class.

**Comment by zxciisdbniel** [comment]
> > > yup. follow the lecture on the maybe context. you should only have one file, Maybe.java
> > 
> > 
> > did you have to redefine the private method signatures of get(), isEmpty() etc to cater for Maybe<?> and Maybe<? extends T>?
> 
> for others who are curious, you shouldnt need to define multiple types of get() or isEmpty(). everything should be within the anonymous inner class.

does that include the public methods like orElse? I can't seem to get my temp Maybe classes to use these methods

**Comment by redcolorbicycle** [comment]
your class should be within an interface, but remember to specify the defined methods within your class as public methods within the interface
